### DISCUSSION TITLE
RS485 Modbus Bridge

### DISCUSSION THREAD

**Discussion by dexterbot80** [arendst/Tasmota#18244] [Q&A]
Hello,
Can someone help me connect the [Carlo Gavazzi](https://www.gavazzionline.com/pdf/EM24DINDS.pdf) energy meter to the tasmota via RS485?
I use [UART TTL to RS485 Two-way Converter ](https://www.elecrow.com/uart-ttl-to-rs485-twoway-converter-p-1545.html )  and [OLIMEX ESP32 GATEWAY board](https://www.tme.com/us/en-us/details/esp32-gateway/development-kits-for-data-transmission/olimex/?gclid=Cj0KCQjw8e-gBhD0ARIsAJiDsaXQqmwe4KrZ1LeTbsTDlJXfX5ClvFl7-bo-r7X6P0jUG1L5x7ecvJEaAhhfEALw_wcB&brutto=1) 
Thank you

**Comment by dexterbot80** [comment]
For example, with this command I should see the voltage, but in order for the measurement to be correct, I must convert from Decimal to Hexadecimal, then **reverse** the result obtained and convert it back to Decimal....
23:43:20.455 CMD: ModBusSend {"deviceaddress": 1, "functioncode": 3, "startaddress": 0, "type":"int32", "count":1}
23:43:20.464 RSL: RESULT = {"ModbusSend":"Done"}
23:43:20.506 RSL: RESULT = {"ModbusReceived":{"DeviceAddress":1,"FunctionCode":3,"StartAddress":0,"Length":9,"Count":1,"Values":[156106752]}}

156106752 Decimal to Hexa  = 94E0000 then i need to reverse this result
000094E Hexa to Decimal = 2382
2382/10 = 238.2 exactly the value shown on the meter display

**Reply by jeroenst** [reply]
Your device has a different endianity, changing endianity is not finished yet.

I will create an option to change endianity in the comming days.

**Comment by jeroenst** [comment]
Implemented in : https://github.com/arendst/Tasmota/pull/18650

**Reply by dexterbot80** [reply]
> Implemented in : #18650

Hello, 
I think that you should be the boss of all Indians :)
Thank you very much @jeroenst Jeroen

**Reply by jeroenst** [reply]
You're welcome!
